connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDataToCatalyst.scala

```diff
@@ -87,22 +79,9 @@ private[sql] case class ProtobufDataToCatalyst(
     mode
   }
 
-  @transient private lazy val nullResultRow: Any = dataType match {
```

Comment:
@LuciferYang, @justaparth moving the discussion about backward compatibility to this comment so that we have a nice thread. I agree with concerns. We have three options to handle this:

  * (A) Do not include option to have old behavior (this PR, **preferred** by author).
      * I think 99.99 % of the cases this is will be either not relevant (because of mode is not set), or does not matter.
      * For the remaining cases I think users will understand and update the query.
 * (B) Provide an option to switch to old behavior. The default is new behavior (**less preferred**).
      * I am ok to to do this option, though I don't think it will be used much.
      * We will have a deprecation warning and will remove this option and relevant code in next major release.
 * (C) Keep old behavior as the default and provide option for new behavior (**least preferred**).
     * I do not think this is a good option.
     * It  makes it hard for users to benefit from the PR.

Reply:
@rangadi I remember that Spark 3.4 already had this feature, but Jira only mentioned 3.5, so does this bug not exist in 3.4? Or should we need to correct `Affects Version/s` and `Fix Version/s` in Jira?

Reply:
> This exists from day 1. Updated 'affects version' to 3.4.0.

I think the `Affects Version/s:` should be `3.4.2`, `3.5.0`, `4.0.0`

then the `Fix Version/s` should be `3.4.3`, `3.5.1` and `4.0.0`?

Reply:
Make sense. Added `3.4.3` as as well. Thanks for updating.